core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -650,6 +646,23 @@ object ConsumerGroupCommand extends Logging {
       }
     }
 
+    private def checkOffsetRange(topicPartition: TopicPartition, shiftedOffset: Long) = {
+      val newOffset = getLogEndOffset(topicPartition) match {
```

Comment:
Also, we don't handle errors in `getLogEndOffset`, which seems a bit unsafe. Maybe we should check for the `Unknown` and `Ignore` cases and raise an error. In general, this is the point I was trying to make: rather than crossing our fingers and trying to go ahead with the reset, It is safer to fail fast on errors.